plugins/module_utils/network/sonic/config/system/system.py

```diff
@@ -370,17 +391,25 @@ def patch_want_with_default(self, want, ac_address_only=False):
     def get_replaced_config(self, have, want):
 
         replaced_config = dict()
+        top_level_want = False
 
-        h_hostname = have.get('hostname', None)
         w_hostname = want.get('hostname', None)
-        if (h_hostname != w_hostname) and w_hostname:
-            replaced_config = have.copy()
-            return replaced_config
-        h_intf_name = have.get('interface_naming', None)
         w_intf_name = want.get('interface_naming', None)
-        if (h_intf_name != w_intf_name) and w_intf_name:
-            replaced_config = have.copy()
-            return replaced_config
+
+        if w_hostname is not None or w_intf_name is not None:
+            top_level_want = True
+
+        if top_level_want:
+            h_hostname = have.get('hostname', None)
+            if (h_hostname != w_hostname):
+                replaced_config = have.copy()
+                return replaced_config
+
+            h_intf_name = have.get('interface_naming', None)
+            if (h_intf_name != w_intf_name):
+                replaced_config = have.copy()
+                return replaced_config
+
```

Comment:
If top_level_want and any of w_hostname or w_intf_name is not equal to one in  have, return replaced_config. it looks like no change from the existing code logically.

Also suppose w_hostname = None, h_hostname != None and w_intf_name == h_intf_name != None, return have as replaced_config, but in this case, should not replaced.

I may be wrong. Please take look.

Reply:
If both attributes were previously specified and one of them is not specified in the current playbook, that is a top level change (reduction in the number of configured top level attributes).

But the previous check wouldn't have caught that case because the "want" value for the top level attribute to be removed was "None".